galsim/table.py

```diff
@@ -280,3 +280,204 @@ def _LookupTable_hash(self):
 _galsim._LookupTable.__eq__ = _LookupTable_eq
 _galsim._LookupTable.__ne__ = lambda self, other: not self.__eq__(other)
 _galsim._LookupTable.__hash__ = _LookupTable_hash
+
+
+class LookupTable2D(object):
+    """
+    LookupTable2D represents a 2-dimensional lookup table to store function values that may be slow
+    to calculate, for which interpolating from a lookup table is sufficiently accurate.  A
+    LookupTable2D is also useful for evaluating periodic 2-d functions given samples from a single
+    period.
+
+    A LookupTable2D representing the function f(x, y) may be constructed from a list or array of `x`
+    values, a list or array of `y` values, and a 2D array of function evaluations at all
+    combinations of x and y values.  For instance:
+
+        >>> x = y = np.arange(5)
+        >>> z = x + y[:, np.newaxis]  # function is x + y, dimensions of z is (5, 5)
+        >>> tab2d = galsim.LookupTable2D(x, y, z)
+
+    To evaluate new function values with the lookup table, use the () operator:
+
+        >>> print tab2d(2.2, 3.3)
+        5.5
+
+    The () operator can also accept lists for the x and y arguments at which to evaluate the
+    LookupTable2D.  In this case, the table is evaluated for all combinations of x and y values and
+    returned as a 2D array with dimensions (len(y), len(x)).
+
+        >>> print tab2d([0, 1], [2, 3, 4])
+        [[ 2.  3.],
+         [ 3.  4.],
+         [ 4.  5.]]
+
+    Finally, if you want to just evaluate the LookupTable2D at a list of x and y points but without
+    evaluating all possible combinations of x and y values, then you can use the `scatter` keyword
+    with the () operator.
+
+        >>> print tab2d([1, 2], [3, 4], scatter=True)
+        [ 4.  6.]
+
+    The default interpolation method is linear.  Other choices for the interpolant are:
+      - 'floor'
+      - 'ceil'
+
+        >>> tab2d = galsim.LookupTable2D(x, y, z, interpolant='floor')
+        >>> tab2d(2.2, 3.3)
+        5.0
+        >>> tab2d = galsim.LookupTable2D(x, y, z, interpolant='ceil')
+        >>> tab2d(2.2, 3.3)
+        7.0
+
+    The `edge_mode` keyword describes how to handle extrapolation beyond the initial input range.
+    Possibilities include:
+      - 'raise': raise an exception.  (This is the default.)
+      - 'wrap': infinitely wrap the initial range in both directions.
```

Comment:
>'zero' seems like another valid edge mode, where extrapolation would just return zeros. Or maybe constant to allow that constant value to be user-specified rather than just 0.
 
Sure.  Added `edge_mode='constant'` and `constant=...` kwarg for user settable constant.